beam/beam/demand/demand.py

```diff
@@ -632,106 +676,137 @@ def get_descendant_warehouses(company: str, warehouse: str) -> list[str]:
+			allocation.key,
+			allocation.demand,
+			allocation.doctype,
+			allocation.company,
+			allocation.parent,
+			allocation.warehouse,
+			allocation.name,
+			allocation.idx,
+			allocation.item_code,
+			allocation.allocated_date,
+			allocation.allocated_date.as_("delivery_date"),
+			allocation.modified,
+			allocation.stock_uom,
+			allocation.allocated_qty,
+			(
+				fn.Coalesce(
+					(
+						Query.from_(demand).select(demand.total_required_qty).where(allocation.demand == demand.key)
+					),
+					0,
+				)
+				- fn.Coalesce(
+					fn.Sum(
+						Query.from_(allocation)
+						.select(allocation.allocated_qty)
+						.where(allocation.demand == allocation.demand)
+					),
+					0,
+				)
+			).as_("net_required_qty"),
+			(
+				Query.from_(demand).select(demand.total_required_qty).where(allocation.demand == demand.key)
+			).as_("total_required_qty"),
+			allocation.status,
+			allocation.assigned,
+			allocation.creation,
+		)
+		.where(allocation.allocated_qty > 0)
+		.orderby(
+			allocation.delivery_date,
+			allocation.idx,
+			allocation.creation,
+			allocation.parent,
+			order=Order.asc,
+		)
+	)
 
-		# if assigned:
-		# 	_filters += f" AND assigned LIKE %{assigned}%"
-
-	demand_query = f"""
-		SELECT
-			d.key,
-			'' AS demand,
-			d.doctype,
-			d.company,
-			d.parent,
-
-			d.warehouse,
-			d.name,
-			d.idx,
-			d.item_code,
-			d.delivery_date AS allocated_date,
-			d.delivery_date,
-
-			d.modified,
-			d.stock_uom,
-			COALESCE(
-				(SELECT SUM(a.allocated_qty) FROM allocation a WHERE a.demand = d.key),
-				0
-			) AS allocated_qty,
-			d.total_required_qty - COALESCE(
-				(SELECT SUM(a.allocated_qty) FROM allocation a WHERE a.demand = d.key),
-				0
-			) AS net_required_qty,
-			d.total_required_qty,
-
-			'' AS status,
-			d.assigned,
-			d.creation
-		FROM demand d
-		WHERE allocated_qty <= 0
-		{d_filters}
-	"""
-
-	allocation_query = f"""
-		SELECT
-			a.key,
-			a.demand,
-			a.doctype,
-			a.company,
-			a.parent,
-
-			a.warehouse,
-			a.name,
-			a.idx,
-			a.item_code,
-			a.allocated_date AS delivery_date,
-			a.allocated_date,
-
-			a.modified,
-			a.stock_uom,
-			a.allocated_qty,
-			COALESCE(
-				(SELECT d.total_required_qty FROM demand d WHERE a.demand = d.key),
-				0
-			) -
-			COALESCE(
-				(SELECT SUM(c.allocated_qty) FROM allocation c WHERE a.demand = c.demand),
-				0
-			) AS net_required_qty,
-			(SELECT d.total_required_qty FROM demand d WHERE a.demand = d.key) AS total_required_qty,
-
-			a.status,
-			a.assigned,
-			a.creation
-		FROM allocation a
-		WHERE allocated_qty > 0
-		{a_filters}
-		ORDER BY delivery_date, idx, creation, parent ASC
-	"""
+	if a_filters:
+		allocation_query = allocation_query.where(*a_filters)
 
 	record_offset = records_per_page * (page - 1)
-	query = (
-		f"{demand_query} UNION ALL {allocation_query} LIMIT {records_per_page} OFFSET {record_offset}"
-	)
+	final_query = demand_query.union(allocation_query).limit(records_per_page).offset(record_offset)
```

Comment:
TO FIX: individual queries works, the union doesn't

Reply:
@fproldan Heather figured out a strategy for this in Check Run where we needed some queries to be configurable

https://github.com/agritheory/check_run/blob/version-14/check_run/check_run/doctype/check_run/check_run.py#L702-L719